lambda/update/correction.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"github.com/ministryofjustice/opg-data-lpa-store/internal/shared"
+	"github.com/ministryofjustice/opg-data-lpa-store/internal/validate"
+	"github.com/ministryofjustice/opg-data-lpa-store/lambda/update/parse"
+	"time"
+)
+
+type Correction struct {
+	DonorFirstNames string
+	DonorLastName   string
+	DonorOtherNames string
+	DonorDob        shared.Date
+	DonorAddress    shared.Address
+	DonorEmail      string
+	LPASignedAt     time.Time
+}
+
+func (c Correction) Apply(lpa *shared.Lpa) []shared.FieldError {
+
+	lpa.Donor.FirstNames = c.DonorFirstNames
```

Comment:
maybe `parse.Optional()` is confusing, e.g., if you didn't pass `/donor/firstNames` in then `validateCorrection` will set `DonorFirstNames = ""`, which applied here will remove the value